diff -ur adobe-plugins.org/pluginsdk/photoshopapi/photoshop/PIGetFileListSuite.h adobe-plugins/pluginsdk/photoshopapi/photoshop/PIGetFileListSuite.h
--- adobe-plugins.org/pluginsdk/photoshopapi/photoshop/PIGetFileListSuite.h	2016-06-06 10:45:38.000000000 -0400
+++ adobe-plugins/pluginsdk/photoshopapi/photoshop/PIGetFileListSuite.h	2016-07-23 10:46:44.711006843 -0400
@@ -84,6 +84,7 @@
 //	Includes.
 //-------------------------------------------------------------------------------
 #include "PIGeneral.h"
+#include <stdbool.h>
 
 #if MSWindows && defined(__PSWorld__)
 #endif
@@ -126,12 +127,12 @@
 typedef FSSpec PIGetFileListHandleType;
 #endif
 
-typedef SPAPI SPErr (*GetFileHandleListProc) (PIActionDescriptor& des, PIGetFileListHandleType* fileSpec, bool recurse);
+typedef SPAPI SPErr (*GetFileHandleListProc) (PIActionDescriptor* des, PIGetFileListHandleType* fileSpec, bool recurse);
 
 /**
 * Gets a list of file handles from \c xPlatFileSpec as above.
 */
-typedef SPAPI SPErr (*GetXPlatFileHandleListProc) (PIActionDescriptor& des, XPlatFileSpec* fileSpec, bool recurse);
+typedef SPAPI SPErr (*GetXPlatFileHandleListProc) (PIActionDescriptor* des, XPlatFileSpec* fileSpec, bool recurse);
 
 /**
 * Gets a list of browser names in the Helper\\Preview In folder.  
@@ -141,7 +142,7 @@
 * read with the @ref PicaActionListSuite "Action List Suite."
 * @returns Non-zero error if failure.
 */
-typedef SPAPI SPErr (*GetBrowserNameListProc) (PIActionDescriptor& des);
+typedef SPAPI SPErr (*GetBrowserNameListProc) (PIActionDescriptor* des);
 
 /**
 * Browses a URL with a browser from the "Preview In" folder; the browser is chosen based on
@@ -167,18 +168,18 @@
 typedef SPAPI SPErr (*GetBrowserFileSpecProc)(uint16 index, SPPlatformFileSpecification* fileSpec);
 typedef SPAPI SPErr (*GetXPlatBrowserFileSpecProc)(uint16 index, XPlatFileSpec* fileSpec);
 
-typedef	SPAPI SPErr (*GetDefaultSystemScriptProc)(int16& script);
+typedef	SPAPI SPErr (*GetDefaultSystemScriptProc)(int16* script);
 
-typedef SPAPI SPErr (*HasDoubleByteInStringProc)(const char* charString, bool& hasDoubleByte);
+typedef SPAPI SPErr (*HasDoubleByteInStringProc)(const char* charString, bool* hasDoubleByte);
 
 typedef SPAPI SPErr (*GetWebTemplateFolderSpecProc)(SPPlatformFileSpecification* fileSpec);
 typedef SPAPI SPErr (*GetXPlatWebTemplateFolderSpecProc)(XPlatFileSpec* fileSpec);
 
-typedef SPAPI SPErr (*GetWebTemplateListProc) (PIActionDescriptor& styleList, bool recurse);
+typedef SPAPI SPErr (*GetWebTemplateListProc) (PIActionDescriptor* styleList, bool recurse);
 
 typedef SPAPI SPErr (*GetWindowsFileTypesProc) (char * fileTypeList, int32 bufferSize);
 
-typedef SPAPI SPErr (*GetMacFileTypesProc) (OSType * typeList, int32 &numTypes, int32 bufferSize);
+typedef SPAPI SPErr (*GetMacFileTypesProc) (OSType * typeList, int32 *numTypes, int32 bufferSize);
 
 typedef SPAPI SPErr (*GetPresetsFolderSpecProc) (SPPlatformFileSpecification *fileSpec);
 typedef SPAPI SPErr (*GetXPlatPresetsFolderSpecProc) (XPlatFileSpec *fileSpec);
@@ -189,7 +190,7 @@
 typedef SPAPI SPErr (*GetChewableFolderSpecProc) (SPPlatformFileSpecification *folderSpec);
 typedef SPAPI SPErr (*GetXPlatChewableFolderSpecProc) (XPlatFileSpec *folderSpec);
 
-typedef SPAPI SPErr (*GetChewableFreeSpaceProc) (int64& freeSpace);
+typedef SPAPI SPErr (*GetChewableFreeSpaceProc) (int64* freeSpace);
 
 //-------------------------------------------------------------------------------
 //	 GetFileList suite version 1.
diff -ur adobe-plugins.org/pluginsdk/photoshopapi/photoshop/PITypes.h adobe-plugins/pluginsdk/photoshopapi/photoshop/PITypes.h
--- adobe-plugins.org/pluginsdk/photoshopapi/photoshop/PITypes.h	2016-06-06 10:45:38.000000000 -0400
+++ adobe-plugins/pluginsdk/photoshopapi/photoshop/PITypes.h	2016-07-23 10:22:11.686980078 -0400
@@ -36,7 +36,7 @@
 #if !defined(_WIN32_WINNT)
 #define _WIN32_WINNT 0x601 // minimal deployment is Windows 7
 #endif
-#include <sdkddkver.h> // for WINVER
+//#include <sdkddkver.h> // for WINVER
 #endif
 
 #if !MSWindows
diff -ur adobe-plugins.org/pluginsdk/photoshopapi/pica_sp/SPProps.h adobe-plugins/pluginsdk/photoshopapi/pica_sp/SPProps.h
--- adobe-plugins.org/pluginsdk/photoshopapi/pica_sp/SPProps.h	2016-06-06 10:45:38.000000000 -0400
+++ adobe-plugins/pluginsdk/photoshopapi/pica_sp/SPProps.h	2016-07-23 09:31:27.600270913 -0400
@@ -28,7 +28,7 @@
  **/
 
 #include "SPTypes.h"
-#include "SPMdata.h"
+#include "SPMData.h"
 #include "SPPiPL.h"
 
 #ifdef __cplusplus
diff -ur adobe-plugins.org/pluginsdk/samplecode/common/includes/PIDLLInstance.h adobe-plugins/pluginsdk/samplecode/common/includes/PIDLLInstance.h
--- adobe-plugins.org/pluginsdk/samplecode/common/includes/PIDLLInstance.h	2016-06-06 10:45:38.000000000 -0400
+++ adobe-plugins/pluginsdk/samplecode/common/includes/PIDLLInstance.h	2016-07-23 11:09:28.695013991 -0400
@@ -16,7 +16,7 @@
 #include "SPPlugs.h"
 #include "SPAccess.h"
 
-#include "PIUSuites.h"
+//#include "PIUSuites.h"
 
 #include <windows.h>
 
diff -ur adobe-plugins.org/pluginsdk/samplecode/common/includes/PIUSuites.h adobe-plugins/pluginsdk/samplecode/common/includes/PIUSuites.h
--- adobe-plugins.org/pluginsdk/samplecode/common/includes/PIUSuites.h	2016-06-06 10:45:38.000000000 -0400
+++ adobe-plugins/pluginsdk/samplecode/common/includes/PIUSuites.h	2016-07-23 11:07:52.449899859 -0400
@@ -25,7 +25,7 @@
 #include "PIGetFileListSuite.h"
 #include "PIBufferSuite.h"
 #include "PIChannelPortsSuite.h"
-#include "JSScriptingSuite.h"
+//#include "JSScriptingSuite.h"
 
 #if __PIWin__
 	#ifndef AliasHandle
diff -ur adobe-plugins.org/pluginsdk/samplecode/common/includes/PIUtilities.h adobe-plugins/pluginsdk/samplecode/common/includes/PIUtilities.h
--- adobe-plugins.org/pluginsdk/samplecode/common/includes/PIUtilities.h	2016-06-06 10:45:38.000000000 -0400
+++ adobe-plugins/pluginsdk/samplecode/common/includes/PIUtilities.h	2016-07-23 11:26:30.818732449 -0400
@@ -2,7 +2,7 @@
 // Copyright  1993 - 2002 Adobe Systems Incorporated
 // All Rights Reserved
 //
-// NOTICE:  Adobe permits you to use, modify, and distribute this 
+// NOTICE:  Adobe permits you to use, modify, and distribute this
 // file in accordance with the terms of the Adobe license agreement
 // accompanying it.  If you have received this file from a source
 // other than Adobe, then your use, modification, or distribution
@@ -16,7 +16,7 @@
 //	Description:
 //		This library contains the headers for the macros, and
 //		routines to simplify the use of suites and also some
-//		helpful common plug-in functions. 
+//		helpful common plug-in functions.
 //
 //	Use:
 //		PIUtilities is intended to group common functions
@@ -24,13 +24,13 @@
 //		plug-in programming.
 //
 //-------------------------------------------------------------------------------
- 
+
 #ifndef __PIUtilities_H__
 #define __PIUtilities_H__
 
 #include <stddef.h>				// Standard definitions.
 
-#include "PITypes.h"			// Photoshop types.
+//#include "PITypes.h"			// Photoshop types.
 #include "PIGeneral.h"			// Photoshop general routines.
 #include "PIActions.h"			// Photoshop scripting.
 #include "PIAbout.h"			// AboutRecord structure.
@@ -45,10 +45,10 @@
 	#include <stdlib.h>
 	#include <winver.h>
 	#include <windowsx.h>
-	#include "PIDLLInstance.h"
+//	#include "PIDLLInstance.h"
 
 	typedef char * StringPtr;
-	
+
 #else // Macintosh or other.
 
 	#define HANDLE Handle
@@ -113,7 +113,7 @@
 extern "C" {
 #endif
 
-    
+
 typedef enum
 {
     kUnknown = -1,
@@ -206,7 +206,7 @@
 		returns int32				Number of bytes free in buffer or 0 if
 									none free or no procs available.
 */
-int32 HostBufferSpace (BufferProcs *procs); 
+int32 HostBufferSpace (BufferProcs *procs);
 
 /**	Returns the amount of space available in the host buffer.
 	@param
@@ -216,7 +216,7 @@
 		returns int64				Number of bytes free in buffer or 0 if
 									none free or no procs available.
 */
-int64 HostBufferSpace64 (BufferProcs *procs); 
+int64 HostBufferSpace64 (BufferProcs *procs);
 
 /** Allocates a buffer of a specified amount.
     Allocate a buffer of the appropriate size setting bufferID to the ID
@@ -235,7 +235,7 @@
 									0 if an error occurred.
 */
 OSErr HostAllocateBuffer (BufferProcs *procs,
-						  const int32 inSize, 
+						  const int32 inSize,
 						  BufferID *outBufferID);
 
 /** Allocates a buffer of a specified amount.
@@ -255,11 +255,11 @@
 									0 if an error occurred.
 */
 OSErr HostAllocateBuffer64 (BufferProcs *procs,
-						  const int64 inSize, 
+						  const int64 inSize,
 						  BufferID *outBufferID);
 
 /** Free the buffer with the given ID.
-	
+
 	@param
 		BufferProcs *procs			Pointer to BufferProcs callback.
 	@param
@@ -288,7 +288,7 @@
 		returns Ptr					Locks the buffer and returns a pointer to
 									it, if successful; NULL if not.
 */
-Ptr HostLockBuffer (BufferProcs *procs, 
+Ptr HostLockBuffer (BufferProcs *procs,
 					const BufferID inBufferID,
 					Boolean inMoveHigh);
 
@@ -311,7 +311,7 @@
 					   const BufferID inBufferID);
 
 /**	Allocates a buffer which is as tall as possible.  It sets the actual
-	height and bufferID parameters if successful. The following routine 
+	height and bufferID parameters if successful. The following routine
     allocates a buffer which is as tall as possible.  It
     takes as parameters, the desired rowBytes, the minimum height, the
     maximum height, and the fraction of the available buffer space to use
@@ -348,7 +348,7 @@
 							   BufferID *outBufferID);
 
 /** Determines whether the AdvanceState callback is available.
-    Check for the advance state procedure and warn if not present.			 
+    Check for the advance state procedure and warn if not present.
 	The macros assume that gStuff is defined somewhere as a pointer
 	to the current interface record.
 
@@ -361,15 +361,15 @@
 /**	Shows an alert if the AdvanceState callback is not available.
 	AlertID #define with the ID of the "Need host" alert.  See header file.
     kNeedVers #define with the ID of the "Need host ^0" string.  See header file.
-	Pops a dialog with the warning "Need host version 3.0" if the 
+	Pops a dialog with the warning "Need host version 3.0" if the
 	proc is not available.
-	
+
 	@param AdvanceStateProc proc		Pointer to AdvanceStateProc.
 	@return TRUE				If the AdvanceState callback is available.
 			FALSE				If the AdvanceState callback is absent.
 */
 Boolean WarnHostAdvanceStateAvailable (AdvanceStateProc proc);
-   
+
 /** Determines whether the ResourceProcs (Pseudo-Resources) callback is available.
 
 	A pseudo-resource is a cross-platform resource that is stored as
@@ -542,7 +542,7 @@
 	@return
 		returns FALSE				If the HandleProcs callback is absent.
 */
-Boolean HostHandleProcsAvailable (HandleProcs *procs, 
+Boolean HostHandleProcsAvailable (HandleProcs *procs,
 								  Boolean *outNewerVersion);
 
 /** Determines whether the HandleProcs callback is available.
@@ -607,7 +607,7 @@
 */
 Handle HostNewHandle (HandleProcs *procs, const size_t inSize);
 
-/** Disposes a handle allocated via the Handle suite. 
+/** Disposes a handle allocated via the Handle suite.
 
 	The HandleProcs are cross-platform master pointers that point to
 	pointers that point to data that is allocated in the Photoshop
@@ -632,7 +632,7 @@
 */
 void HostDisposeHandle (HandleProcs *procs, Handle h);
 
-/** Returns the size of a handle. 
+/** Returns the size of a handle.
 
 	The HandleProcs are cross-platform master pointers that point to
 	pointers that point to data that is allocated in the Photoshop
@@ -657,7 +657,7 @@
 */
 size_t HostGetHandleSize (HandleProcs *procs, Handle h);
 
-/** Resizes a handle. 
+/** Resizes a handle.
 
 	The HandleProcs are cross-platform master pointers that point to
 	pointers that point to data that is allocated in the Photoshop
@@ -681,11 +681,11 @@
 		returns OSErr				noErr if able to resize, otherwise
 									whatever error occurred (if any).
 */
-OSErr HostSetHandleSize (HandleProcs *procs, 
-						 Handle h, 
+OSErr HostSetHandleSize (HandleProcs *procs,
+						 Handle h,
 						 const size_t inNewSize);
 
-/** Locks a handle, returning a pointer to the data. 
+/** Locks a handle, returning a pointer to the data.
 
 	The HandleProcs are cross-platform master pointers that point to
 	pointers that point to data that is allocated in the Photoshop
@@ -716,7 +716,7 @@
 									procs are unavailable.
 */
 Ptr HostLockHandle (HandleProcs *procs,
-					Handle h, 
+					Handle h,
 					const Boolean inMoveHigh);
 
 /** Unlocks a handle.
@@ -762,7 +762,7 @@
 
 	@param
 		const Handle inHandle		Data to concatenate onto outHandle.
-		
+
 	@return
 		returns OSErr				Error checking is pretty loose
 									on this one, since it will create
@@ -787,7 +787,7 @@
 									will create a new handle.
 	@param
 		const Handle inHandle		Data to concatenate onto outHandle.
-		
+
 	@return
 		returns OSErr				Error checking is pretty loose
 									on this one, since it will create
@@ -796,7 +796,7 @@
 OSErr HostHandleCopy (HandleProcs *procs,
 					  Handle *outHandle,
 					  const Handle inHandle);
-					  
+
 /** Reads a resource from a Windows or Macintosh plug-in.
 
 	@param
@@ -913,7 +913,7 @@
 */
 Handle HostPString2Handle (HandleProcs *procs,
 						  const Str255 inString);
-						  
+
 /** Copies a handle's contents to a pascal string with a maximum length of 255.
 
 	The HandleProcs are cross-platform master pointers that point to
@@ -973,12 +973,12 @@
 
 /**	Shows an alert if the DisplayPixels callback is not available.
 	Pops a dialog with the warning "Need host version 2.5.2." if the
-	proc is not available. 
+	proc is not available.
 	AlertID #define with the ID of the "Need host" alert.  See header file.
 	kNeedVers #define with the ID of the "Need host ^0" string.  See header file.
 
 	@param DisplayPixelsProc proc		Pointer to DisplayPixelsProc.
-	
+
 	@return TRUE If the DisplayPixels callback is available.
 	        FALSE If the DisplayPixels callback is absent.
 */
@@ -1017,13 +1017,13 @@
 
 	AlertID						#define with the ID of the "Need host"
 								alert.  See header file.
-	
+
 	kNeedVers					#define with the ID of the "Need host ^0"
 								string.  See header file.
 
 	kWrongHost					#define with the ID of the "Wrong host"
 								string.  See header file.
-	
+
 	PropertyProcs *procs		Pointer to PropertyProcs callback.
 
 	@return
@@ -1186,7 +1186,7 @@
 OSErr	HostCloseReader (PIDescriptorParameters *procs, HandleProcs *hProcs, PIReadDescriptor *token);
 
 /** Closes a write token, stores its handle in the global parameter block for
-	the host to use, sets the token to NULL, and sets the recordInfo to 
+	the host to use, sets the token to NULL, and sets the recordInfo to
 	plugInDialogOptional (the default).
 
 	The Descriptor Parameters suite are callbacks designed for
@@ -1242,12 +1242,12 @@
 		returns OSErr					noErr or error if one occurred.
 */
 OSErr HostPutObj (PIDescriptorParameters *procs,
-				  HandleProcs *hProcs, 
+				  HandleProcs *hProcs,
 				  PIWriteDescriptor token,
 				  DescriptorKeyID key,
 				  DescriptorTypeID type,
 				  PIDescriptorHandle *h);
-				  
+
 /** Interrogates playInfo and returns whether host has told plug-in to pop
 	its dialog.
 
@@ -1267,7 +1267,7 @@
 Boolean HostPlayDialog (PIDescriptorParameters *procs);
 
 /** Determines whether the ColorServices callback is available.
-	@param 
+	@param
 	    ColorServicesProc proc		Pointer to ColorServicesProc.
 
 	@return
@@ -1287,7 +1287,7 @@
 									string.  See header file.
 	@param
 		ColorServicesProc proc		Pointer to ColorServicesProc.
-	
+
 	@return
 		returns TRUE				If the ColorServices callback is available.
 		returns FALSE				If the ColorServices callback is absent.
@@ -1315,7 +1315,7 @@
 		int16 *outColor				Array of 4 int16s set to inColors.
 */
 OSErr CSSetColor (int16 *outColor,
-				  const int16 inColor1, 
+				  const int16 inColor1,
 				  const int16 inColor2,
 				  const int16 inColor3,
 				  const int16 inColor4);
@@ -1617,8 +1617,8 @@
 									"^0", "^1", and/or "^2".
 */
 void PIParamText(unsigned char *s,
-				 unsigned char *r0, 
-				 unsigned char *r1, 
+				 unsigned char *r0,
+				 unsigned char *r1,
 				 unsigned char *r2);
 
 /** Raises a given base to a certain power.
@@ -1661,7 +1661,7 @@
 		size_t							Length of new string.
 
 	Truncation can be detected in this manner:
-	
+
 	if (PIstrlcat(pname, file, sizeof(pname)) >= sizeof(pname))
 		goto toolong;
 
@@ -1719,9 +1719,9 @@
 		returns int16				Exact match was found, from char
 									outMatch to end of string.
 */
-int16 HostMatch(const char *inSearch, 
-				const char *inTarget, 
-				const Boolean ignoreCase, 
+int16 HostMatch(const char *inSearch,
+				const char *inTarget,
+				const Boolean ignoreCase,
 				const int16 inLength);
 
 /** Finds an occurrence of a target string in a source string.  Note that
@@ -1751,7 +1751,7 @@
 		returns itn16				Exact match was found, from char
 									outMatch to end of string.
 */
-int16 HostStringMatch(Str255 s1, 
+int16 HostStringMatch(Str255 s1,
 					  const char *s2,
 					  const Boolean ignoreCase);
 
@@ -1796,8 +1796,8 @@
 										valid if both pointers were
 										valid.
 */
-void HostBlockMove (Ptr outTarget, 
-					const Ptr inSource, 
+void HostBlockMove (Ptr outTarget,
+					const Ptr inSource,
 					const size_t inLength);
 
 /** Copies a chunk of memory. This is by no means the most effective way
@@ -1817,10 +1817,10 @@
 										valid if both pointers were
 										valid.
 */
-void HostCopy (void *outTarget, 
+void HostCopy (void *outTarget,
 			   const void *inSource,
 			   const size_t inLength);
-			   
+
 /**	Clears a string by zeroing its first two bytes. Takes care of Pascal and C strings.
 	Note: This will automatically get inlined on non-debug builds.
 	@param unsigned char *string Any string, at least two bytes.
@@ -1834,7 +1834,7 @@
 
 #define PIMatch(inSearch, inTarget) \
 	HostMatch(inSearch, inTarget, true, PIstrlen(inSearch))
-	
+
 #define PISMatch(inSearch, inTarget) \
 	HostStringMatch(inSearch, inTarget, true)
 
@@ -1930,9 +1930,9 @@
 */
 Ptr AllocateGlobals(void * resultAddr,
 					void * paramBlockAddr,
-					HandleProcs *procs, 
-					const size_t size, 
-					intptr_t *data, 
+					HandleProcs *procs,
+					const size_t size,
+					intptr_t *data,
 					FProcP InitGlobals);
 
 //-------------------------------------------------------------------------------
@@ -1992,7 +1992,7 @@
 
 #define WarnAdvanceStateAvailable() \
 	WarnHostAdvanceStateAvailable (gStuff->advanceState)
-	
+
 #define AdvanceState() \
 	(*(gStuff->advanceState)) ()
 
@@ -2002,19 +2002,19 @@
 
 #define ResourceProcsAvailable(outNewerVersion) \
 	HostResourceProcsAvailable (gStuff->resourceProcs, outNewerVersion)
-	
+
 #define WarnResourceProcsAvailable() \
 	WarnHostResourceProcsAvailable (gStuff->resourceProcs)
-	
+
 #define CountPIResources(type) \
 	HostCountPIResources (gStuff->resourceProcs, type)
-	
+
 #define GetPIResource(type,index) \
 	HostGetPIResource (gStuff->resourceProcs, type, index)
-	
+
 #define DeletePIResource(type,index) \
 	HostDeletePIResource (gStuff->resourceProcs, type, index)
-	
+
 #define AddPIResource(type,data) \
 	HostAddPIResource (gStuff->resourceProcs, type, data)
 
@@ -2024,49 +2024,49 @@
 
 #define HandleProcsAvailable(outNewerVersion) \
 	HostHandleProcsAvailable (gStuff->handleProcs, outNewerVersion)
-	
+
 #define WarnHandleProcsAvailable() \
 	WarnHostHandleProcsAvailable (gStuff->handleProcs)
-	
+
 #define PINewHandle(size) \
 	HostNewHandle (gStuff->handleProcs, size)
-	
+
 #define PIDisposeHandle(h) \
 	HostDisposeHandle (gStuff->handleProcs, h)
-	
+
 #define PIGetHandleSize(h) \
 	HostGetHandleSize (gStuff->handleProcs, h)
-	
+
 #define PISetHandleSize(h,size) \
 	HostSetHandleSize (gStuff->handleProcs, h, size)
-	
+
 #define PILockHandle(h,moveHigh) \
 	HostLockHandle (gStuff->handleProcs, h, moveHigh)
-	
+
 #define PIUnlockHandle(h) \
 	HostUnlockHandle (gStuff->handleProcs, h)
 
 #define PIHandleCat(h1, h2)	\
 	HostHandleCat(gStuff->handleProcs, h1, h2)
-	
+
 #define PIHandleCopy(h1, h2) \
 	HostHandleCopy(gStuff->handleProcs, h1, h2)
 
 #define PICString2Handle(s) \
 	HostCString2Handle (gStuff->handleProcs, s)
-	
+
 #define PIHandle2CString(h, s, max) \
 	HostHandle2CString (gStuff->handleProcs, h, s, max)
 
 #define PIPString2Handle(s) \
 	HostPString2Handle (gStuff->handleProcs, s)
-	
+
 #define PIHandle2PString(h, s) \
 	HostHandle2PString (gStuff->handleProcs, h, s)
-	
+
 #define PIHandle2VersionString(h, s) \
 	HostHandle2VersionString (gStuff->handleProcs, h, s)
-	
+
 #define PIHandle2Version(h, major, minor, fix) \
 	HostHandle2Version (gStuff->handlProcs, major, minor, fix)
 
@@ -2098,20 +2098,20 @@
 
 #define PutSimple(key, simple)  \
 	PISetProp('8BIM', key, 0, simple, nil)
-	
+
 #define GetComplex(key, index, complex)	\
 	PIGetProp('8BIM', key, index, 0, complex)
-	
+
 #define PutComplex(key, index, complex)	\
 	PISetProp('8BIM', key, index, 0, complex)
-	
+
 //-------------------------------------------------------------------------------
 //	ChannelPortProcs -- Macro definitions
 //-------------------------------------------------------------------------------
 
 #define ChannelPort \
 	(gStuff->channelPortProcs)
-	
+
 #define ReadPixels \
 	(ChannelPort->readPixelsProc)
 
@@ -2126,7 +2126,7 @@
 
 #define WarnChannelPortAvailable() \
 	WarnHostChannelPortAvailable (ChannelPort)
-	
+
 //-------------------------------------------------------------------------------
 //	Error reporting function -- Macro definitions
 //-------------------------------------------------------------------------------
@@ -2164,13 +2164,13 @@
 
 #define	OpenReader(array) \
 	OpenReadDesc(DescParams->descriptor, array)
-	
+
 #define CloseReadDesc(token) \
 	Reader->closeReadDescriptorProc(token)
 
 #define CloseReader(token) \
 	HostCloseReader(DescParams, gStuff->handleProcs, token)
-	
+
 #define OpenWriter() \
 	Writer->openWriteDescriptorProc()
 
@@ -2184,13 +2184,13 @@
 
 #define PIGetKey(token, key, type, flags) \
 	Reader->getKeyProc(token, key, type, flags)
-	
+
 #define PIGetEnum(token, value)	\
 	Reader->getEnumeratedProc(token, value)
-				
+
 #define PIPutEnum(token, key, type, value) \
 	Writer->putEnumeratedProc(token, key, type, value)
-	
+
 #define PIGetInt(token, value) \
 	Reader->getIntegerProc(token, value)
 
@@ -2199,16 +2199,16 @@
 
 #define PIPutInt(token, key, value) \
 	Writer->putIntegerProc(token, key, value)
-	
+
 #define PIGetFloat(token, value) \
 	Reader->getFloatProc(token, value)
-	
+
 #define PIGetPinFloat(token, min, max, value) \
 	Reader->getPinnedFloatProc(token, min, max, value)
 
 #define PIPutFloat(token, key, value) \
 	Writer->putFloatProc(token, key, value)
-	
+
 #define PIGetUnitFloat(token, unit, value) \
 	Reader->getUnitFloatProc(token, unit, value)
 
@@ -2217,7 +2217,7 @@
 
 #define PIPutUnitFloat(token, key, unit, value) \
 	Writer->putUnitFloatProc(token, key, unit, value)
-	
+
 #define PIGetBool(token, value) \
 	Reader->getBooleanProc(token, value)
 
@@ -2226,13 +2226,13 @@
 
 #define PIGetText(token, value) \
 	Reader->getTextProc(token, value)
-	
+
 #define PIPutText(token, key, value) \
 	Writer->putTextProc(token, key, value)
-	
+
 #define PIGetAlias(token, value) \
 	Reader->getAliasProc(token, value)
-	
+
 #define PIPutAlias(token, key, value) \
 	Writer->putAliasProc(token, key, value)
 
@@ -2244,16 +2244,16 @@
 
 #define PIGetClass(token, value) \
 	Reader->getClassProc(token, value)
-	
+
 #define PIPutClass(token, key, value) \
 	Writer->putClassProc(token, key, value)
-	
+
 #define PIGetRef(token, value) \
 	Reader->getSimpleReferenceProc(token,value)
-	
+
 #define PIPutRef(token, key, value) \
 	Writer->putSimpleReferenceProc(token, key, value)
-	
+
 #define PIGetObj(token, type, value) \
 	Reader->getObjectProc(token, type, value)
 
@@ -2262,16 +2262,16 @@
 
 #define PIPutObjProc(token, key, type, value) \
 	Writer->putObjectProc(token, key, type, value)
-	
+
 #define PIGetCount(token, value) \
 	Reader->getCountProc(token, value)
-	
+
 #define PIPutCount(token, key, value) \
 	Writer->putCountProc(token, key, value)
-	
+
 #define PIGetStr(token, value) \
 	Reader->getStringProc(token, value)
-	
+
 #define PIPutStr(token, key, value) \
 	Writer->putStringProc(token, key, value)
 
@@ -2284,7 +2284,7 @@
 
 #define WarnColorServicesAvailable() \
 	WarnHostColorServicesAvailable (gStuff->colorServices)
-	
+
 #define ColorServices(info) \
 	(*(gStuff->colorServices)) (info)
 
@@ -2302,6 +2302,7 @@
 } // End of extern "C" block.
 #endif // __cplusplus
 
+#ifdef __cplusplus
 /// Swap from big endian to little endian
 template <class Type>
 void Swap(Type & in)
@@ -2320,6 +2321,7 @@
 		back--;
 	}
 }
+#endif // __cplusplus
 
 void PStrCat(const StringPtr src, StringPtr dst);
 void PStrInsert(StringPtr src, StringPtr dst);
@@ -2370,22 +2372,22 @@
 
 inline int32 Convert_size_t_to_int32(size_t x)
 	{
-	return static_cast<int32> (x);
+	return (int32) (x);
 	}
 
 inline uint32 Convert_size_t_to_uint32(size_t x)
 	{
-	return static_cast<uint32> (x);
+	return (uint32) (x);
 	}
 
 inline int Convert_size_t_to_int(size_t x)
 	{
-	return static_cast<int> (x);
+	return (int) (x);
 	}
 
 inline unsigned int Convert_size_t_to_uint(size_t x)
 	{
-	return static_cast<unsigned int> (x);
+	return (unsigned int) (x);
 	}
 
 #else
@@ -2395,29 +2397,29 @@
 
 #define Convert_size_t_to_int(x)	(static_cast<int> (x))				// no need to safety check on 32 bit compile. simply silence warnings
 #define Convert_size_t_to_uint(x)	(static_cast<unsigned int> (x))		// no need to safety check on 32 bit compile. simply silence warnings
-	
+
 #endif
 
 // 16 bits
 
 inline int16 Convert_size_t_to_int16(size_t x)
 	{
-	return static_cast<int16> (x);
+	return (int16) (x);
 	}
 
 inline uint16 Convert_size_t_to_uint16(size_t x)
 	{
-	return static_cast<uint16> (x);
+	return (uint16) (x);
 	}
 
 inline char Convert_size_t_to_char(size_t x)
 	{
-	return static_cast<char> (x);
+	return (char) (x);
 	}
 
 inline unsigned char Convert_size_t_to_uchar(size_t x)
 	{
-	return static_cast<unsigned char> (x);
+	return (unsigned char) (x);
 	}
 
 /******************************************************************************/
